src/CacheFactory.php

```diff
@@ -42,126 +29,28 @@ protected function createWithConfig(ContainerInterface $container, string $confi
             throw OutOfBoundsException::forMissingConfigKey('class');
         }
 
-        $instance = null;
+        $cache = $container->has($config['class']) ? $container->get($config['class']) : new $config['class']();
```

Comment:
Since we are no longer supporting the various cache instance types, `new $config['class']` only remains useful/relevant for something like a `NullCache`: we can probably always use `$container->get()` for all scenarios.

Reply:
also for old default cache implementation - 'array'. So its usefull to simplify migration by add config:
```
  'doctrine' => [
    'cache' => [
      'array' => ['class' => \Symfony\Component\Cache\Adapter\ArrayAdapter::class],
    ],
  ],
```
without forcing adding useless services into container

Reply:
This would be a sufficient approach, but a void cache is preferable.

While it may be tempting to use `array` as a default, it is risky to do so for caches other than DQL and metadata caches, because `array` can break in long-running processes.

For result caches, a void/null cache would be best as a default.

Reply:
I would let clients decide which caches to use for which purposes. Also caches are optional, so we can avoid create own implementation of void/null/blackhole cache or be locked to another vendor (e.g. symfony/cache).

Above in https://github.com/Roave/psr-container-doctrine/pull/72#discussion_r1098482535 its just example how simple could be migration for client who use our default config with 'array' cache and if we will left constructor call in `CacheFactory`. We have no dependency on `symfony/cache`.